src/zkevm_specs/evm/instruction.py

```diff
@@ -328,40 +378,40 @@ def mul_word_by_u64(self, multiplicand: RLC, multiplier: Expression) -> Tuple[RL
 
         return self.rlc_encode(product_bytes), FQ(quotient_hi)
 
-    def rlc_to_fq_unchecked(self, word: RLC, n_bytes: int) -> Tuple[FQ, FQ]:
-        return self.bytes_to_fq(word.le_bytes[:n_bytes]), self.is_zero(
-            self.sum(word.le_bytes[n_bytes:])
+    def mul_add_words(self, a: RLC, b: RLC, c: RLC, d: RLC) -> FQ:
+        """
+        The function constrains a * b + c == d, where a, b, c, d are 256-bit words.
+        It returns the overflow part of a * b + c.
+        """
+        a64s = self.word_to_64s(a)
+        b64s = self.word_to_64s(b)
+        c_lo, c_hi = self.word_to_lo_hi(c)
+        d_lo, d_hi = self.word_to_lo_hi(d)
+
+        t0 = a64s[0] * b64s[0]
+        t1 = a64s[0] * b64s[1] + a64s[1] * b64s[0]
+        t2 = a64s[0] * b64s[2] + a64s[1] * b64s[1] + a64s[2] * b64s[0]
+        t3 = a64s[0] * b64s[3] + a64s[1] * b64s[2] + a64s[2] * b64s[1] + a64s[3] * b64s[0]
+        carry_lo = (t0 + (t1 * 2**64) + c_lo - d_lo) / (2**128)
+        carry_hi = (t2 + (t3 * 2**64) + c_hi + carry_lo - d_hi) / (2**128)
+        overflow = (
+            carry_hi
+            + a64s[1] * b64s[3]
+            + a64s[2] * b64s[2]
+            + a64s[3] * b64s[1]
+            + a64s[2] * b64s[3]
+            + a64s[3] * b64s[2]
+            + a64s[3] * b64s[3]
         )
 
-    def rlc_to_fq_exact(self, word: RLC, n_bytes: int) -> FQ:
-        if any(word.le_bytes[n_bytes:]):
-            raise ConstraintUnsatFailure(f"Word {word} has too many bytes to fit {n_bytes} bytes")
-
-        return self.bytes_to_fq(word.le_bytes[:n_bytes])
-
```

Comment:
ditto

Reply:
This function is not removed, but just renamed into `rlc_to_fq`.